### PR TITLE
swiftinterface grab bag

### PR DESCRIPTION
**Pull request by graydon**
Some fixes to diagnostics and a renamed test dir. Nothing dramatic, just figured I'd try to file things in smaller increments this time :)

### CODE REVIEW

**Comment by graydon** [comment]
@swift-ci please test

File: lib/Frontend/ParseableInterfaceSupport.cpp [hunk 1]
```diff
@@ -265,6 +265,9 @@ static bool buildSwiftModuleFromSwiftInterface(
     for (auto const &Dep : DepNames) {
       auto DepStatus = FS.status(Dep);
       if (!DepStatus) {
+        Diags.diagnose(SourceLoc(),
+                       diag::missing_dependency_of_parseable_module_interface,
```
Comment (jrose-apple) [inline lib/Frontend/ParseableInterfaceSupport.cpp]: Ah, I think it's worth printing the error that occurred too; the dependency might not actually be missing.

**Review by jrose-apple — COMMENTED** [review]
Looks good, needs tests. :-)

File: lib/Frontend/ParseableInterfaceSupport.cpp (hunk 1 above)
Comment (graydon) [reply lib/Frontend/ParseableInterfaceSupport.cpp]: Fixed.

**Comment by graydon** [comment]
@swift-ci please test and merge

**Comment by graydon** [comment]
@swift-ci please test and merge

**Comment by graydon** [comment]
@swift-ci please test

**Comment by graydon** [comment]
@swift-ci please test Linux platform

**Comment by graydon** [comment]
@swift-ci please test Linux platform

**Comment by graydon** [comment]
@swift-ci please test Linux platform

**Comment by graydon** [comment]
@swift-ci please test

**Comment by graydon** [comment]
@swift-ci please test and merge

**Comment by graydon** [comment]
@swift-ci please test Linux platform

**Comment by graydon** [comment]
@swift-ci please test Linux platform

**Comment by graydon** [comment]
subsumed by #20458, closing.

**Pull request closed without merging**